lib/graphql/batch/mutation_execution_strategy.rb

```diff
@@ -0,0 +1,19 @@
+module GraphQL::Batch
+  class MutationExecutionStrategy < GraphQL::Query::SerialExecution
+    class FieldResolution < GraphQL::Query::SerialExecution::FieldResolution
+      def get_finished_value(raw_value)
+        context = execution_context.query.context
+        old_execution_strategy = context.execution_strategy
+        begin
+          context.execution_strategy = GraphQL::Batch::ExecutionStrategy.new
+          raw_value = GraphQL::Batch::Promise.resolve(raw_value).sync
```

Comment:
IIUC this line can go above the setting of the execution strategy and therefore outside the `begin/ensure` block; I think I'd prefer that just to make it super-obvious which lines need the new execution strategy (specifically, the `super` call)